### DISCUSSION TITLE
캐릭터 불러오기가 안돼요..

### DISCUSSION THREAD

**Discussion by almondsky** [naverz/zepeto-studio-kor#825] [4. General]
안녕하세요, 
캐릭터 불러오기에서 헤매다 어떤 분께서 저와 같은 상황으로 여기에 글을 올린것을 봤습니다. 그 분께 해결책으로 제시해주신 스크립 코드를 똑같이 아래와 같이 복사 붙여넣기 해서 넣어봤구요.
<img width="885" alt="Screen Shot 2022-05-31 at 2 44 31 PM" src="https://user-images.githubusercontent.com/105544091/171101809-5002840d-67f6-485a-9e79-b87c0cb74bda.png">

이렇게 넣어서 저장한 후 유니티에서 실행해보면 캐릭터가 뜨지않고 
<img width="807" alt="Screen Shot 2022-05-31 at 2 44 14 PM" src="https://user-images.githubusercontent.com/105544091/171101483-b3255e55-8341-4100-be53-12b9568e1dff.png">
이 문구만 자꾸 뜨네요.. 캐릭터 소환은 어떻게 하는건가요? ㅠ 

추가적으로 스크립 작성시 이 부분 "[ZEPETO_ID]"에는 "[현재 내 아이디 넣기]"를 해야하나요 아니면 '[ZEPETO_현재 내아이디 넣기]"를 해야하는 건가요?

**Comment by pumfit** [comment]
"[ZEPETO_ID]"에는 제페토 아이디를 넣어주면 됩니다. 만약 작성자분의 제페토 아이디가 almondsky라면 "[ZEPETO_ID]"해당 부분에 "almondsky"와 같이 넣어주시면 될 것 같습니다!

**Reply by almondsky** [reply]
답변감사합니다. 아이디 부분에 제페토 아이디 넣고 저 위에 스크립트대로 실행해봤는데.. typescript sources is not compiled yet. 이라고 뜨는데 ㅠㅠ 혹시 어디가 잘못되었는지 아실까요?

**Reply by Zepeto-TechEvangelist** [reply]
안녕하세요, 코드에 오류가 있어서 컴파일이 되지 않은 상황이므로 전체 코드를 보여주시겠어요?

**Reply by almondsky** [reply]
안녕하세요, 답변 너무너무 감사드려요. 전체 코드라 하시면 이거 말씀하시는거 맞죠? 이렇게 입력해서
<img width="875" alt="Screen Shot 2022-05-31 at 4 10 28 PM" src="https://user-images.githubusercontent.com/105544091/171113659-c5443429-563b-4e9d-9b12-8e778032cde0.png">
실행시켜보니 저렇게 오류가 떠요.

**Reply by Zepeto-TechEvangelist** [reply]
1. import { ZepetoScriptBehaviour } from 'ZEPETO.Script' 가 한 줄에 나오도록 코드 수정
2. 아이디 부분에 [] 제거 해서 다시 실행해 보세요~ 💪

**Reply by almondsky** [reply]
말씀해주신대로 이렇게 수정했구요
<img width="768" alt="Screen Shot 2022-05-31 at 4 23 48 PM" src="https://user-images.githubusercontent.com/105544091/171116106-0fb1609e-bb9e-4353-afad-5b81f43b92de.png">
실행해보니 이렇게 떠요.. 
<img width="858" alt="Screen Shot 2022-05-31 at 4 24 41 PM" src="https://user-images.githubusercontent.com/105544091/171116204-df130c50-f8a1-466e-b2e2-e4312e5a1840.png">
 저 파란부분이 캐릭터가 떠야하는데 어디서 오류가 지금 나서 저렇게 뜨는건가요...? ㅠ

**Reply by Zepeto-TechEvangelist** [reply]
스크립트에 오류가 없고 스크립트가 오브젝트에 컴포넌트로 추가되어 있다면
캐릭터가 나타나야 정상입니다.
아래 가이드에서 월드 세팅, 프로젝트 세팅, 로그인 부분에서 누락된 부분은 없는지 한번 더 확인 부탁 드립니다.
https://studio.zepeto.me/kr/guides/preparing-for-world-creating

**Reply by almondsky** [reply]
답변 정말 너무 감사합니다. 스크립트는 동일하고 월드 세팅 아래와 같이 등록 및 유니티에도 이렇게 세팅했구요.
<img width="313" alt="Screen Shot 2022-05-31 at 5 02 07 PM" src="https://user-images.githubusercontent.com/105544091/171123638-64c385a7-48e7-461d-9660-91e917cff130.png">
플레이어 세팅 했구, 로그인 까지 진행했구 
<img width="605" alt="Screen Shot 2022-05-31 at 5 04 08 PM" src="https://user-images.githubusercontent.com/105544091/171124069-dbc38c95-0348-4a74-ac72-7464ea50170b.png">

이렇게 씬 등록까지 마친후에 다시 플레이 버튼을 눌러봤더니 화면은 위와 같이 똑같습니다..  
<img width="1565" alt="Screen Shot 2022-05-31 at 5 06 36 PM" src="https://user-images.githubusercontent.com/105544091/171124283-4748b271-fb7e-4990-ac30-b737d5da7097.png">

**Reply by opper9** [reply]
typescript sources is not compiled yet << 저도 이 문구가 계속 나왔는데 유니티랑 에디터 다 지우고 처음부터 다시 했더니 더이상 안뜹니다. 무슨 이유인지 알고싶긴 하네요 ㅠ

**Reply by Zepeto-TechEvangelist** [reply]
@opper9 typescript sources is not compiled yet 는 코드 문법이나 사용성 오류이기 때문에 코드를 전체적으로 공유해주시면 도움을 드릴 수 있습니다.

**Reply by almondsky** [reply]
전 typescrip sources is not compiled yet 은 더이상 안뜨고 플레이 버튼을 누르면 캐릭터 대신 저 파란색 도형만 뜨는데 
코드는 아까와 동일하며, 아래 공유합니다.
<img width="985" alt="Screen Shot 2022-05-31 at 6 24 48 PM" src="https://user-images.githubusercontent.com/105544091/171140571-be009a2e-0452-4b4d-81c3-bf626df432bf.png">

**Reply by Zepeto-TechEvangelist** [reply]
@almondsky 님 코드대로 실행하면 아래와 같이 정상적으로 캐릭터가 보여집니다. 

<img width="337" alt="스크린샷 2022-05-31 오후 6 32 05" src="https://user-images.githubusercontent.com/100118650/171142742-65d3fd8f-0438-4252-8174-a3684a6f90d7.png">

**Reply by almondsky** [reply]
evangelist 님 감사합니다. 해결했습니다. 아마 두가지 씬에 같은 월드 아이디를 연동시켜놨던게 문제였지 않나 싶습니다. 이렇게 도와주셔서 너무 감사합니다!

**Reply by opper9** [reply]
> @opper9 typescript sources is not compiled yet 는 코드 문법이나 사용성 오류이기 때문에 코드를 전체적으로 공유해주시면 도움을 드릴 수 있습니다.

아하 그렇군요 ㅠㅠ코딩을 처음 해봐서 어려운 게 많네요 친절히 알려주셔서 감사합니다

**Reply by opper9** [reply]
<img alt="Screen Shot 2022-05-31 at 4 24 41 PM" width="858" src="https://user-images.githubusercontent.com/105544091/171116204-df130c50-f8a1-466e-b2e2-e4312e5a1840.png">

죄송하지만 이거 파란거 나오는 거 해결방법이 뭔가요?
